Bottom: b962dcf4eecd1c4ce0f1d60698eff7adf337312a
Top:    40162b03f0d0298c368821e998f6b7b54781a942
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-04-05 22:29:51 +0100

Allow PolygonHoles() to be passed a NULL region

Passing a NULL region implies we wish to be called back for all
holes in the polygon, as we don't have a region to test against.

Fixes crash in the DRC window when the design has polygons, and
is viewed with either thindraw, or thindraw polygons.


---

diff --git a/src/polygon.c b/src/polygon.c
index bc7b555..d5df073 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1256,8 +1256,9 @@ PolygonHoles (PolygonType *polygon, const BoxType *range,
     return 0;
   for (pl = pa->contours->next; pl; pl = pl->next)
     {
-      if (pl->xmin > range->X2 || pl->xmax < range->X1 ||
-          pl->ymin > range->Y2 || pl->ymax < range->Y1)
+      if (range != NULL &&
+          (pl->xmin > range->X2 || pl->xmax < range->X1 ||
+           pl->ymin > range->Y2 || pl->ymax < range->Y1))
         continue;
       if (callback (pl, user_data))
         {
